Add instructions for installing on Windows through WSL

jtsymon opened the pull request:
Moving instructions for working with WSL to README.md.
Some stuff to look at:
- Automatically create `browserpass-linux.bat` with `install.ps1` to simplify the instructions? (it can just sit there unused normally)
- If your GPG key has a password (which it really should), the host running under WSL won't be able to unlock it since it can't prompt the user for the password. This means you can't get decrypt any passwords unless you've already got the key loaded in gpg-agent.
Could possibly handle this by prompting for the master password in the browser extension in this case? Seems kinda sketchy though, especially for such an edge case.

jtsymon commented:
Apparently the PR title doesn't link issues 😞 
#192

maximbaz commented:
Thanks!

1. Extending the `install.ps1` sounds like a good idea, do you want to try implementing it? I won't be able to test this bit personally, but we have a couple of Windows users, I can ping them.
2. Yeah I don't think it's a good idea to implement pin-entry dialog in the extension itself, don't want to touch users' passwords 🙂 But it is a very important part, or people will be confused why nothing is working. Could you maybe mention this in the README as well? Just mention that unfortunately there is such issue, and as a workaround it is required to manually unlock the key, e.g. just show a password inside WSL.

maximbaz commented:
By the way, how come on WSL both windows and linux host applications are needed? Are they both being used? The `.bat` file only references the linux host app...

jtsymon commented:
The Windows host application isn't actually needed, but the config files and registry keys are used.
Normally it'd work like
`Browser -> Native Messaging config -> Host Application`
But this does
`Browser -> Native Messaging config (Windows) -> Host Application (Linux)`
So we need the config files from the Windows host application (but not the actual host application), and the Linux host application (but not its config files).

maximbaz commented:
Huh, makes sense now, thanks!

jtsymon commented:
Oh I just noticed another problem. My original `.bat` file assumes the Linux host application is in `~/browserpass/`, but that actually depends on where the user put the browserpass files when installing the Linux host application. So the `.bat` file installed with the Windows host application doesn't actually know where to find the Linux host application.
`install.ps1` could actually also install the Linux version under WSL (behind a prompt or flag), but that could get pretty messy (the cleanest option is probably a separate "WSL" zip that contains the Windows config with the Linux host application and a different install script, but that creates extra maintenance work for very few users).
... so maybe we should scrap the "automatically create the `.bat` file" idea, and just say "create a batch file that uses WSL to load the Linux browserpass host application."

maximbaz commented:
Agreed, let's go with a simple option then. If there will be a public demand for automating then, we will revisit :wink: 

Add the description of the problem with password-protected key and a known workaround, and let's merge it.

jtsymon commented:
I added the note about password-protected keys, and also added a note about _why_ to use this (hopefully keeps people from accidentally using this when they just want a Windows setup).

jtsymon commented:
Only problem is now its taking up half the README for an edge case setup 😆

maximbaz commented on README.md:
```diff
@@ -39,6 +39,22 @@ All release files are signed with [this PGP key](https://keybase.io/maximbaz). T
+
+3. Change the path in `%localappdata%\browserpass\browserpass-firefox.json` (or `-chrome.json`) to point to `browserpass-wsl.bat`
+
+If your GPG key has a password, the host application running under WSL won't be able to unlock it since it can't interactively prompt for the password. This means you can't get decrypt any passwords unless you've already got the key loaded in gpg-agent.
```
typo `get decrypt` -> `decrypt`

maximbaz commented on README.md:
```diff
@@ -39,6 +39,22 @@ All release files are signed with [this PGP key](https://keybase.io/maximbaz). T
+3. Change the path in `%localappdata%\browserpass\browserpass-firefox.json` (or `-chrome.json`) to point to `browserpass-wsl.bat`
+
+If your GPG key has a password, the host application running under WSL won't be able to unlock it since it can't interactively prompt for the password. This means you can't get decrypt any passwords unless you've already got the key loaded in gpg-agent.
+As a workaround, you can use the key (`pass website.com`) in a WSL terminal to load the key into gpg-agent. Then browserpass will work until gpg-agent times out.
```
Maybe add this hint in the end: `(it is possible to configure larger timeouts, check manual for gpg-agent).`

maximbaz commented:
That is totally fine, and I don't see it as an edge case, I like the idea behind WSL and I hope it gets better / more widespread in the future 😉

maximbaz commented:
Cool, thanks a lot - merging in a moment.
